tests/components/sonos/test_media_browser.py

```diff
@@ -95,3 +97,147 @@ async def test_build_item_response(
         browse_item.children[1].media_content_id
         == "x-file-cifs://192.168.42.10/music/The%20Beatles/Abbey%20Road/03%20Something.mp3"
     )
+
+
+async def test_browse_media_root(
+    hass: HomeAssistant,
+    soco_factory: SoCoMockFactory,
+    async_autosetup_sonos,
+    soco,
+    discover,
+    hass_ws_client: WebSocketGenerator,
+) -> None:
+    """Test the async_browse_media method."""
+
+    client = await hass_ws_client()
+    await client.send_json(
+        {
+            "id": 1,
+            "type": "media_player/browse_media",
+            "entity_id": "media_player.zone_a",
+        }
+    )
+    response = await client.receive_json()
+    assert response["success"]
+    assert len(response["result"]["children"]) == 2
+    favorites = {
+        "title": "Favorites",
+        "media_class": "directory",
+        "media_content_type": "favorites",
+        "media_content_id": "",
+        "can_play": False,
+        "can_expand": True,
+        "thumbnail": "https://brands.home-assistant.io/_/sonos/logo.png",
+        "children_media_class": None,
+    }
+    assert favorites in response["result"]["children"]
+
+    music_library = {
+        "title": "Music Library",
+        "media_class": "directory",
+        "media_content_type": "library",
+        "media_content_id": "",
+        "can_play": False,
+        "can_expand": True,
+        "thumbnail": "https://brands.home-assistant.io/_/sonos/logo.png",
+        "children_media_class": None,
+    }
+    assert music_library in response["result"]["children"]
```

Comment:
```suggestion
    expected = [{
        "title": "Favorites",
        "media_class": "directory",
        "media_content_type": "favorites",
        "media_content_id": "",
        "can_play": False,
        "can_expand": True,
        "thumbnail": "https://brands.home-assistant.io/_/sonos/logo.png",
        "children_media_class": None,
    },
    {
        "title": "Music Library",
        "media_class": "directory",
        "media_content_type": "library",
        "media_content_id": "",
        "can_play": False,
        "can_expand": True,
        "thumbnail": "https://brands.home-assistant.io/_/sonos/logo.png",
        "children_media_class": None,
    }]
    assert response["result"]["children"] == unordered(expected)
```

You also need the following import `from pytest_unordered import unordered`

Use only unordered if the order is not guaranteed

Reply:
Thank you - that is much simpler.  The order that is returned to the media_browser should be the same order that they are received from Sonos - as the Sonos data is already sorted (e.g. by artist, album, etc.).

I've updated all the new tests in this module to use this technique.

Reply:
I think maybe it would make sense to use snapshot testing for comparing such big lists.

Reply:
Was not thinking about snapshot but you are right :) @PeteRager Would it possible to use the snapshot asserting instead :) It makes the test easier to read, and I apologize that I was not thinking about it.

Reply:
Sure, I've made the changes, waiting for CI to run, it's been queued for a while now.